CCO-784: update to pick up tls-min-version and tls-cipher-suites

jstuever opened the pull request:
This change merges in recent upstream changes to pick up tls-min-version and tls-cipher-suite flags. It also updates to using golang v1.25 to be compatible with recent upstream changes.

eks-distro-pr-bot committed:
Creating PR to update Go version to 1.24.11 (#288)

2uasimojo commented:
NB: I think we also want `tls-cipher-suites`; and eventually something about "curves" once https://github.com/openshift/api/pull/2583 lands.

Ganiredi committed:
1.35.0 dependency update (#291)

jstuever commented:
This is currently a cherry-pick of the commits posted upstream. Ideally, the upstream PR will merge and we can pull them in directly. If that doesn't happen in time for this release, we may have to carry them until it does.

jstuever committed:
feat: add flags tls-cipher-suites and tls-min-version (#292)

* feat: add flag to configure minimum TLS version for webhook server

This commit introduces a new command-line flag `--tls-min-version` to the
webhook server. This allows callers to specify the minimum supported TLS
version (1.0, 1.1, 1.2, or 1.3), improving security flexibility by allowing
the enforcement of stricter TLS protocols.

Assisted-by: gemini-3-pro-preview

* feat: add flag to configure TLS cipher suites

This commit introduces a new command-line flag `-tls-cipher-suites` to the
webhook server. This allows users to specify a comma-separated list of
allowed TLS cipher suites, enhancing security configurability.

Assisted-by: gemini-3-pro-preview

jstuever commented:
/assign @dlom

dlom commented:
/lgtm

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-aws-ipi-private-shared-vpc-phz-sts-mini-perm-f7

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-aws-ipi-wavelength-sts-fips-mini-perm-f14

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-azure-mag-ipi-fips-f7

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-azure-ipi-vmgenv1-mini-perm-f7

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-gcp-ipi-marketplace-mini-perm-f28

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-gcp-ipi-disc-priv-oidc-oc-mirror-fips-f28-destructive

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-azure-ipi-disc-oidc-mini-perm-f14

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-ibmcloud-ipi-mini-perm-f28

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-nutanix-ipi-data-disks-f28

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-vsphere-ipi-amd-f7-netobserv

jstuever committed:
feat: tls-min-version to also accept tls constant names (#296)

This commit enables the ability to specify the tls-min-version using the
names of the constants in crypto/tls/common.go (i.e., VersionTLS10,
VersionTLS11, VersionTLS12, VersionTLS13).

Assisted-by: gemini-3-pro-preview

jstuever committed:
Merge upstream latest

jstuever committed:
update vendor

jstuever committed:
Update to golang-1.25

dlom commented:
/lgtm

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-aws-ipi-private-shared-vpc-phz-sts-mini-perm-f7

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-nutanix-ipi-data-disks-f28

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-vsphere-ipi-amd-f7-netobserv

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-ibmcloud-ipi-mini-perm-f28

huangmingxia commented:
/payload-job periodic-ci-openshift-openshift-tests-private-release-4.22-amd64-nightly-azure-ipi-disc-oidc-mini-perm-f14

huangmingxia commented:
/verified by @huangmingxia 
For test results, see comments in [CCO-784](https://issues.redhat.com/browse/CCO-784).
Cluster installation regression tests were verified through the CI runs above.
